docs/decisions/0015-i18n-support.rst

```diff
@@ -0,0 +1,46 @@
+15. Internationalization support.
+---------------------------------
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+Paragon's components are extensively used in applications that support i18n.
+Currently consumers of Paragon need to handle i18n themselves, i.e. pass translated
+strings to components as props. Furthermore, some of the components have hardcoded strings in them which do not
+allow translation in any way which can cause a lot of headache for consumers.
+
+Decision
+--------
+
+Paragon will natively support internationalization. We will use `react-intl <https://formatjs.io/docs/react-intl/>`_ library
+to translate all string type props and move all hardcoded strings from components to respective props.
+We will setup Transifex pipeline to auto pull new translations.
+All translated messages will be exported from Paragon to allow consumers to use them throughout their apps.
+
+Consequences
+------------
+
+- This will be a breaking change for all consumers who do not support i18n in their apps since react-intl requires a specific context to be used in the app. All apps (even if they don't intend use i18n) that use Paragon's components **must** use `IntlProvider` component to provide i18n context to Paragon's components, most common and recommended usage is to wrap root component of the app:
```

Comment:
Might be good to talk about how this would be used in conjunction with `frontend-platform/i18n` as well. For example, most MFEs that consume Paragon and frontend-platform, use `frontend-platform`'s `AppProvider` component, which renders the `IntlProvider` for you. Will this be compatible?